storage/integration_test/src/integration_test.cc

```diff
@@ -386,6 +387,102 @@ const std::string kSimpleTestFile =
     "pariatur. Excepteur sint occaecat cupidatat non proident, sunt in "
     "culpa qui officia deserunt mollit anim id est laborum.";
 
+// TODO(almostmatt): Make this test only run on desktop.
+// Set a test-only flag to enable retrying 404s.
+TEST_F(FirebaseStorageTest, TestReadBytesWithMaxRetryDuration) {
+  int shorter_duration_sec = 2;
+  int longer_duration_sec = 6;
+  SignIn();
+
+  // Attempt to read a non-existent file, and verify that GetBytes fails.
+  {
+    LogDebug("Call GetBytes on a nonexistent ref.");
+    firebase::storage::StorageReference ref =
+        CreateFolder().Child("File1.txt");
+    LogDebug("Storage URL: gs://%s%s", ref.bucket().c_str(),
+            ref.full_path().c_str());
+    cleanup_files_.push_back(ref);
+
+    const size_t kBufferSize = 1024;
+    char buffer[kBufferSize];
+    memset(buffer, 0, sizeof(buffer));
```

Comment:
If we're not checking that buffer stays all 0s, any reason to memset it?

Reply:
No reason, removed this line.